.github/workflows/release.yml

```diff
@@ -1,9 +1,16 @@
 name: Build distribution
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+      - main
+    tags:
+      - "20*" # Push events to matching 20*, i.e. 2022.2.0, 2099.8.1
```

Comment:
We intentionally build the packages every time to ensure packaging works as expected, so I'd rather not restrict this.

If you look at the Publish step you'll see we have an `if` there that means it will only push to PyPI when tags are created.